spec/models/event_spec.rb

```diff
@@ -0,0 +1,90 @@
+require "rails_helper"
+
+RSpec.describe Event, type: :model do
+  describe "relationships" do
+    it { is_expected.to belong_to :venue }
+    it { is_expected.to belong_to :category }
+    it { is_expected.to have_many :ticket_types }
+  end
+
+  describe "validations" do
+    it { is_expected.to validate_presence_of :name }
+    it { is_expected.to validate_presence_of :extended_html_description }
+    it { is_expected.to validate_presence_of :venue }
+    it { is_expected.to validate_presence_of :category }
+    it { is_expected.to validate_presence_of :starts_at }
+    it { is_expected.to validate_uniqueness_of(:name).scoped_to(:venue_id, :starts_at) }
+  end
+
+  describe "starts_at_to_s" do
+    let(:start_time) { DateTime.new(2016, 7, 7, 8, 0, 0) }
+    let(:event) { create(:event, starts_at: start_time) }
+    let(:expected_date_string) { "Thursday, 07 Jul 2016  8:00 AM" }
+
+    it "returns starts_at as string with full date format" do
+      expect(event.starts_at_to_s).to eq expected_date_string
+    end
+  end
+
+  describe "starts_at_to_month" do
+    let(:start_time) { DateTime.new(2016, 7, 7, 8, 0, 0) }
+    let(:event) { create(:event, starts_at: start_time) }
+    let(:expected_date_string) { "July" }
+
+    it "returns starts_at as string with full month format" do
+      expect(event.starts_at_to_month).to eq expected_date_string
+    end
+  end
+
+  describe "starts_at_to_day" do
+    let(:start_time) { DateTime.new(2016, 7, 7, 8, 0, 0) }
+    let(:event) { create(:event, starts_at: start_time) }
+    let(:expected_date_string) { "07" }
+
+    it "returns starts_at as string with day format" do
+      expect(event.starts_at_to_day).to eq expected_date_string
+    end
+  end
+
+  describe "region_name" do
+    let(:event) { create(:event) }
+    let(:expected_region) { event.venue.region.name }
+
+    it "returns region of this event" do
+      expect(event.region_name).to eq expected_region
```

Comment:
This is the problem with using exact same code in test and in code. Not a good practice. Find a way ;)